### PR TITLE
KAFKA-15288: Change BrokerApiVersionsCommandTest to support kraft mode

### PR DESCRIPTION
**Pull request by vveicc**
Change BrokerApiVersionsCommandTest to support kraft mode.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public static NodeApiVersions create() {
      */
     public static NodeApiVersions create(Collection<ApiVersion> overrides) {
         List<ApiVersion> apiVersions = new LinkedList<>(overrides);
-        for (ApiKeys apiKey : ApiKeys.zkBrokerApis()) {
+        for (ApiKeys apiKey : ApiKeys.clientApis()) {
```
Comment (dengziming) [inline clients/src/main/java/org/apache/kafka/clients/NodeApiVersions.java]: This method is used in too many places so we should be careful about this change, we'd better keep it unchanged in this PR, and do this change in another patch.

File: core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala [hunk 2]
```diff
@@ -17,37 +17,47 @@
...
-      props.setProperty(KafkaConfig.ControlPlaneListenerNameProp, "CONTROLLER")
-      props.setProperty(KafkaConfig.ListenerSecurityProtocolMapProp, "CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT")
-      props.setProperty("listeners", "PLAINTEXT://localhost:0,CONTROLLER://localhost:0")
-      props.setProperty(KafkaConfig.AdvertisedListenersProp, "PLAINTEXT://localhost:0,CONTROLLER://localhost:0")
-      props.setProperty(KafkaConfig.UnstableApiVersionsEnableProp, "true")
-      props
-    }).map(KafkaConfig.fromProps)
+    if (isKRaftTest()) {
+      TestUtils.createBrokerConfigs(1, null).map(props => {
+        props.setProperty(KafkaConfig.UnstableApiVersionsEnableProp, "true")
```
Comment (dengziming) [inline core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala:45]: Please add a comment about this change, I guess this is related to KIP-848, and we can remove this after KIP-848 is all set.

File: core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala [hunk 3]
```diff
@@ -56,21 +66,30 @@ class BrokerApiVersionsCommandTest extends KafkaServerTestHarness {
     assertTrue(lineIter.hasNext)
     assertEquals(s"${bootstrapServers()} (id: 0 rack: null) -> (", lineIter.next())
     val nodeApiVersions = NodeApiVersions.create
```
Comment (dengziming) [inline core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala]: We should change how we construct `NodeApiVersions` here,  such as `val nodeApiVersions = new NodeApiVersions(clientApis.map(ApiVersionsResponse.toApiVersion).asJava, Collections.emptyList(), false)`

File: core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala [hunk 4]
```diff
@@ -56,21 +66,30 @@ class BrokerApiVersionsCommandTest extends KafkaServerTestHarness {
...
-      val terminator = if (apiKey == enabledApis.last) "" else ","
+        val terminator = if (apiKey == clientApis.last) "" else ","
 
-      val line = s"\t${apiKey.name}(${apiKey.id}): $versionRangeStr [usable: $usableVersion]$terminator"
-      assertTrue(lineIter.hasNext)
-      assertEquals(line, lineIter.next())
+        val line = s"\t${apiKey.name}(${apiKey.id}): $versionRangeStr [usable: $usableVersion]$terminator"
+        assertTrue(lineIter.hasNext)
+        assertEquals(line, actual)
+      }
```
Comment (dengziming) [inline core/src/test/scala/integration/kafka/admin/BrokerApiVersionsCommandTest.scala:97]: We should add a else branch to do similar assertions, such as 
```
        val line = s"\t${apiKey.name}(${apiKey.id}): UNSUPPORTED,"
        assertTrue(lineIter.hasNext)
        assertEquals(line, actual)
```

**Review by dengziming — COMMENTED** [review]
Thank you for this PR @vveicc , I left some comments, PTAL.

**Comment by vveicc** [comment]
Thank you for your advice @dengziming, this is very helpful to me. I have made some changes, PTAL.
